lib/pigeon/dispatcher_worker.ex

```diff
@@ -43,4 +43,23 @@ defmodule Pigeon.DispatcherWorker do
         {:stop, reason, %{adapter: adapter, state: new_state}}
     end
   end
+
+  @impl GenServer
+  def handle_call(:info, _from, %{adapter: adapter, state: state}) do
+    info = %{peername: peername(state)}
+    {:reply, info, %{adapter: adapter, state: state}}
+  end
+
+  defp peername(state) do
+    with %{socket: socket} <- state,
+         %{connection: connection} <- :sys.get_state(socket),
+         %{config: %{socket: socket2}} <- :sys.get_state(connection),
+         %{socket: socket3} <- :sys.get_state(socket2),
+         {_, {_, port, _, _}, _} <- socket3,
+         {:ok, addr} <- :inet.peername(port) do
```

Comment:
this could be combined into one step if you really want, since the socket that is internal to kadabra state (`socket2` in this code) is a `:ssl` socket type. You could just call `:ssl.peername(socket3)` to get the remote address. 

We handle both possible ways the socket could be structured (though I think it will always be an ssl socket for our use case) in the troubleshooting module [here](https://github.com/Tango-Tango/ptt/blob/master/apps/troubleshooting/lib/troubleshooting/push_notifications.ex#L149-L155)